examples/plotting/file/graphs.py

```diff
@@ -0,0 +1,55 @@
+from bokeh.plotting import figure, gridplot, GridSpec, output_file, show
+import networkx as nx
+from sympy import *
+
+def graph_draw(g, layout=nx.circular_layout, node_color="white", text_color="black"):
+    pos = layout(g)
+    labels = [ str(v) for v in g.nodes() ]
+    vx, vy = zip(*[ pos[v] for v in g.nodes() ])
+    xs, ys = [], []
+    for (a, b) in g.edges():
+        x0, y0 = pos[a]
+        x1, y1 = pos[b]
+        xs.append([x0, x1])
+        ys.append([y0, y1])
+    f = figure(width=300, height=300,
+               x_axis_type=None, y_axis_type=None,
+               outline_line_color=None,
+               tools=[], toolbar_location=None)
+    f.multi_line(xs, ys, line_color="black")
+    f.circle(vx, vy, size=16, line_color="black", fill_color=node_color)
+    f.text(vx, vy, text=labels, text_color=text_color,
+           text_font_size="10px", text_align="center", text_baseline="middle")
```

Comment:
When working on this example, a few things bothered me and should be considered for future development. I presume some are known and there might be even issues open for them, but I wanted to list issues that are specific to this particular piece of code in one place. This example btw. took me way too much time to create than I would expect given the higher-level API. I know we could have graph chart implemented, but still we won't have all of them (or at least not now), so we should be able to do more with less typing and thinking using plotting API.

1. Constructing grid plots should more flexible. That's partially fixed in this PR.
2. `figure()` should allow `bare` mode (and maybe more), so that I don't have to disable each feature individually. Pretty much we should have different sets of defaults ready to be used. This is at least partially different from theming (disabling toolbar vs. disabling (changing color of) outline) and very different from usual notion of (property) defaults in bokeh.
3. There should be a way to compose glyphs, so that I could create e.g. encircled text. This may or may not have something to do with annotations. Most likely not. Additionally, e.g. hit testing would work only for entire group.
4. Even besides composite glyphs, centering text as we do it is cumbersome (but maybe that's the right way given issues with computing text dimensions on canvas).
5. There should a way to say that I want certain e.g. line color for a group of glyphs (this is similar yet not equivalent to (3)). I imagine this being configured via context managers.
6. Glyph functions should be more flexible about their inputs (mostly coordinates), so that I don't have to spend half of a function for rearranging data. Accepting points as `(x, y)` pairs is a minimum. At this point I'm not sure how the API will look like (lack of function overriding makes it a bit difficult, given existing API flexibility).